src/libsyntax_ext/proc_macro_registrar.rs

```diff
@@ -232,6 +190,126 @@ impl<'a> Visitor<'a> for CollectCustomDerives<'a> {
             };
             self.handler.span_err(item.span, msg);
         }
+    }
+
+    fn collect_custom_attribute(&mut self, item: &'a ast::Item, attr: &'a ast::Attribute) {
```

Comment:
`collect_custom_attribute` isn't clear to me -- how about `collect_attr_proc_macro`?